app/controllers/concerns/table_on_steroids/table_concern.rb

```diff
@@ -36,168 +38,185 @@ module TableConcern
+        objects = all_column_search(objects, columns_on_steroid, params[:search], table_on_steroids) unless global_search
       end
 
-      [:activerecord, :array].each do |t|
-
-        #column search 
-        objects = objects_where(objects, columns_on_steroid, t) 
+      %i[activerecord array].each do |t|
+        # column search
+        objects = objects_where(objects, columns_on_steroid, t)
 
-        #apply filters
+        # apply filters
         if params[:filters].present?
-          params[:filters].each_pair do | k, v |
+          params[:filters].each_pair do |k, v|
             filter = columns_on_steroid[k]
             next unless filter.present? && filter[t].present?
+
             objects = filter[t][:filter_lambda].call(objects, v)
           end
         end
 
-        #order
-        if params[:knowledge] && params[:knowledge] && params[:knowledge][:order].present? && (object_order = columns_on_steroid[params[:knowledge][:order]]).present?       
-          if(object_order[t] && object_order[t][:order_lambda])
-            objects = objects.reorder(nil) if(objects.is_a?(ActiveRecord::Base) || objects.is_a?(ActiveRecord::Relation))
+        # order
+        if params[:knowledge] && params[:knowledge] && params[:knowledge][:order].present? && (object_order = columns_on_steroid[params[:knowledge][:order]]).present?
+          if object_order[t] && object_order[t][:order_lambda]
+            objects = objects.reorder(nil) if objects.is_a?(ActiveRecord::Base) || objects.is_a?(ActiveRecord::Relation)
             objects = object_order[t][:order_lambda].call(objects)
           end
-        elsif  ( object_order = columns_on_steroid.select{ |c,v| v[t] && v[t][:default_order] }.first&.last).present?
-          if(object_order[t] && object_order[t][:order_lambda])  
-            objects = objects.reorder(nil) if(objects.is_a?(ActiveRecord::Base) || objects.is_a?(ActiveRecord::Relation))
+        elsif (object_order = columns_on_steroid.select { |_c, v| v[t] && v[t][:default_order] }.first&.last).present?
+          if object_order[t] && object_order[t][:order_lambda]
+            objects = objects.reorder(nil) if objects.is_a?(ActiveRecord::Base) || objects.is_a?(ActiveRecord::Relation)
             objects = object_order[t][:order_lambda].call(objects)
           end
-        elsif(objects.is_a?(ActiveRecord::Base) || objects.is_a?(ActiveRecord::Relation))
-          #objects = objects.order('created_at desc')
+        elsif objects.is_a?(ActiveRecord::Base) || objects.is_a?(ActiveRecord::Relation)
+          # objects = objects.order('created_at desc')
         end
-
       end
 
-      #pagination 
+      # pagination
       return (include_counts ? [objects, 1, objects.count] : objects) if all_pages
-      if(objects.is_a?(ActiveRecord::Base) || objects.is_a?(ActiveRecord::Relation))
-        objects = objects.page(params[:page]).per(OBJECTS_PER_PAGE) 
+
+      unpaginated_objects = objects
```

Comment:
We have the full recordset or array here, can we do something like the following to pull the index and calculate the page?

in the passed in table_on_steroids config, you have the key for active record or array:
global_active_record_search_key = table_on_steroids&.dig(:global_active_record_search_key) || :id
global_array_search_key_index = table_on_steroids&.dig(:global_array_search_key_index) || 0
then
activerecord: objects.pluck(global_active_record_search_key).index(fetch_object_id)
array: objects.map {|obj| obj[global_array_search_key_index]}.index(fetch_object_id)